React Dynamic Of Todos Created

Reptiloid044 opened the pull request:
[DEMO LINK](https://Reptiloid044.github.io/react_dynamic-list-of-todos/)
Hi!

rboiar commented on src/App.tsx:
```diff
@@ -1,14 +1,47 @@
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
```
I don't see the loader when the page is loading

```suggestion
  const [isLoading, setIsLoading] = useState(true);
```

rboiar reviewed (changes requested):
Good job, but fix the loader ;)

rboiar reviewed (approved):
Nice!

dellachaise commented on src/App.tsx:
```diff
@@ -1,14 +1,47 @@
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [queryFilter, setQueryFilter] = useState('');
+  const [selectStatus, setSelectStatus] = useState('all');
```
Better create enum for filter type and also use it in `swith case`.

dellachaise commented on src/App.tsx:
```diff
@@ -1,14 +1,47 @@
+    }
+  });
+
+  const visibleTodos = filterBy.filter(todoItem => {
```
Now you twice filter todos. Can you filter just one time?

dellachaise commented on src/App.tsx:
```diff
@@ -17,18 +50,36 @@ export const App: React.FC = () => {
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todos={todos}
```
For what purpose do you need `todos` in `TodoFilter`?

dellachaise commented on src/App.tsx:
```diff
@@ -17,18 +50,36 @@ export const App: React.FC = () => {
+                <TodoList
+                  todos={visibleTodos}
+                  selectedTodoId={todoId}
+                  selectedTodo={(todo: React.SetStateAction<number>) => setTodoId(todo)}
```
```suggestion
                  selectedTodo={setTodoId}
```

dellachaise commented on src/App.tsx:
```diff
@@ -17,18 +50,36 @@ export const App: React.FC = () => {
+        <TodoModal
+          todoId={todoId}
+          todos={visibleTodos}
+          selectedTodo={(todo: React.SetStateAction<number>) => setTodoId(todo)}
```
```suggestion
          selectedTodo={setTodoId}
```

dellachaise commented on src/components/TodoFilter/TodoFilter.tsx:
```diff
@@ -1,34 +1,69 @@
+            value={selectStatus}
+            onChange={handleSelectStatus}
+          >
+            <option value="all">All</option>
```
You could import here the previously created enum and use it.

dellachaise commented on src/components/TodoFilter/TodoFilter.tsx:
```diff
@@ -1,34 +1,69 @@
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {query && (
+            // eslint-disable-next-line jsx-a11y/control-has-associated-label
```
try to add aria-label attribute to buttons all over the project to avoid using `eslint-disable`.

dellachaise commented on src/components/TodoModal/TodoModal.tsx:
```diff
@@ -1,12 +1,42 @@
+  selectedTodo: (value: number) => number | void;
+};
+
+// eslint-disable-next-line max-len
```
Why do you need this `eslint-disable`?

dellachaise commented on src/components/TodoModal/TodoModal.tsx:
```diff
@@ -1,12 +1,42 @@
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(userFromServer => setUser(userFromServer));
```
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code

dellachaise reviewed (changes requested):
Good job 👍 

Just add changes according to comments 👇

Reptiloid044 commented:
Hi Dasha!
Thank you for the comments, I have fixed according to your recommendation, please check=)
Added ENAM;
Removed double filtration;
Used Async\await for useEffect;
Added aria-label attribute to buttons;

AlinaMatuschak commented on src/components/TodoList/TodoList.tsx:
```diff
@@ -1,100 +1,78 @@
+        {todos.map(({ id, title, completed }) => (
+          <tr
+            data-cy="todo"
+            className=""
```
```suggestion
```

AlinaMatuschak commented on src/components/TodoList/TodoList.tsx:
```diff
@@ -1,100 +1,78 @@
+              )}
+            </td>
+            <td className="is-vcentered is-expanded">
+              <p className={completed
+                ? 'has-text-success'
+                : 'has-text-danger'}
```
use classnames

AlinaMatuschak reviewed (changes requested):
<img width="527" alt="Screenshot 2022-10-03 at 20 37 33" src="https://user-images.githubusercontent.com/47606379/193642614-111e1f2a-de7a-4266-ba49-0f0b7e7fabb9.png">

Reptiloid044 commented:
Hi Alina!
I have added filter by title and classNames for TodoList, check please=)

AlinaMatuschak commented on src/App.tsx:
```diff
@@ -1,14 +1,50 @@
+    fetchData();
+  }, []);
+
+  const filteredTodos = todos.filter(({ title, completed }) => {
+    switch (filterBy) {
+      case FilterType.Active:
+        return !completed && titleCheck(title, queryFilter);
+
+      case FilterType.Completed:
+        return completed && titleCheck(title, queryFilter);
+
+      default:
+        return true;
+    }
+  });
```
when filterBy = 'all' you don't check includes queryFilter in the title

Reptiloid044 replied:
I have added the default check for 'All'

AlinaMatuschak replied:
Right, but on this default check you didn’t check query at all )
